neonvm/controllers/virtualmachine_controller.go

```diff
@@ -1401,3 +1513,44 @@ func getEnvVarValue(envVarName string) (string, error) {
 	}
 	return value, nil
 }
+
+func sshKeygen() ([]byte, []byte, error) {
+	publicKey, privateKey, err := ed25519.GenerateKey(rand.Reader)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	publicKeyBytes, err := encodePublicKey(publicKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	privateKeyBytes, err := encodePrivateKey(privateKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return publicKeyBytes, privateKeyBytes, nil
+}
+
+func encodePrivateKey(privateKey ed25519.PrivateKey) ([]byte, error) {
+	privBlock, err := ssh.MarshalPrivateKey(privateKey, "")
+	if err != nil {
+		return nil, err
+	}
+	privatePEM := pem.EncodeToMemory(privBlock)
+
+	return privatePEM, nil
+}
+
+func encodePublicKey(publicKey ed25519.PublicKey) ([]byte, error) {
+	sshPublicKey, err := ssh.NewPublicKey(publicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	pubKeyBytes := ssh.MarshalAuthorizedKey(sshPublicKey)
+	return pubKeyBytes, nil
+}
+
+func ptr[T any](t T) *T { return &t }
```

Comment:
couple things about this function

1. elsewhere in the controllers package, it seems like instead of using e.g. `ptr[T](...)` we use `&[]T{...}[0]` — maybe just go with that instead of this function? (although in the future, it'd probably be better to use a function for this in all the places)
2. this is defined quite far from any other usages - if we keep, let's relocate?

Reply:
Removed the `ptr` function and used the array hack in 837a8fc1fc93a32fd12ef2a780dafa9d40fd208c. Will do the improvements in #733.